File: hello-dyn-java/AdvancedCalculator.java
// 高级计算器实现
public class AdvancedCalculator implements Calculator {
    @Override
    public double addFn(double a, double b) {
        System.out.printf("[ADVANCED] 加法: %.2f + %.2f%n", a, b);
        return a + b;
    }
    
    @Override
    public double subFn(double a, double b) {
        System.out.printf("[ADVANCED] 减法: %.2f - %.2f%n", a, b);
        return a - b;
    }
    
    @Override
    public double mulFn(double a, double b) {
        System.out.printf("[ADVANCED] 乘法: %.2f * %.2f%n", a, b);
        return a * b;
    }
    
    @Override
    public double divFn(double a, double b) {
        if (b == 0) {
            throw new ArithmeticException("除数不能为零");
        }
        System.out.printf("[ADVANCED] 除法: %.2f / %.2f%n", a, b);
        return a / b;
    }
    
    // 额外功能
    public double powFn(double base, double exponent) {
        return Math.pow(base, exponent);
    }
}

File: hello-dyn-python/advanced_calculator.py
from calculator_base import Calculator
import math


class AdvancedCalculator(Calculator):
    """高级计算器实现"""

    def addFn(self, a, b):
        print(f"[ADVANCED] 加法: {a:.2f} + {b:.2f}")
        return a + b

    def subFn(self, a, b):
        print(f"[ADVANCED] 减法: {a:.2f} - {b:.2f}")
        return a - b

    def mulFn(self, a, b):
        print(f"[ADVANCED] 乘法: {a:.2f} * {b:.2f}")
        return a * b

    def divFn(self, a, b):
        if b == 0:
            raise ValueError("除数不能为零")
        print(f"[ADVANCED] 除法: {a:.2f} / {b:.2f}")
        return a / b

    def powFn(self, base, exponent):
        """幂运算"""
        return math.pow(base, exponent)
